src/arrays/arraymath.jl

```diff
@@ -121,6 +121,11 @@ function \{B<:OrthonormalBasis}(op::AbstractQuMatrix{B}, vec::AbstractQuVector{B
     return QAT(div, bases(op,1))
 end
 
+# dot product of QuArray's
+dot{B<:OrthonormalBasis}(vec1::AbstractQuVector{B}, vec2::AbstractQuVector{B}) = dot(coeffs(vec1), coeffs(vec2))
```

Comment:
+1, we shouldn't allow `DualVector`s as arguments to `dot` - Julia's convention for `dot` is that it takes in two column vectors and applies the transpose/conjugate transpose as necessary.